internal/daemon/daemon.go

```diff
@@ -566,24 +582,12 @@ func (d *Daemon) State() *state.State {
 		Database:       d.db,
 		Remotes:        d.trustStore.Remotes,
 		StartAPI:       d.StartAPI,
-		Stop:           d.Stop,
+		Stop:           d.stop,
```

Comment:
@masnax i was thinking about how we can avoid the need of exporting the `ShutdownDoneCh` in this struct, and avoid the need for the caller of `stopErr := state.Stop()` from having to pass the `stopErr` to `state.ShutdownDoneCh` as it seems rather circular.

Instead how about we do change `state.Stop` to be `func() (exit func(), stopErr error)`.

Then in `shutdownPost` we can call `exit, stopErr := state.Stop()` and then rather than posting to `state.ShutdownDoneCh` here we just call `exit()` which then internally will send the stopErr returned to `d.shutdownDoneCh` (making the channel an internal implementation detail and not part of the API).